ratel/src/astgen/expression.rs

```diff
@@ -269,6 +269,31 @@ impl<'ast> SerializeInLoc for Expression<'ast> {
 
 #[cfg(test)]
 mod test {
+    #[test]
+    fn test_void_expression() {
+        expect_parse!("[1,]", {
+            "type":"Program",
+            "body":[
+                {
+                    "type":"ExpressionStatement",
+                    "expression":{
+                        "type":"ArrayExpression",
+                        "elements":[
+                            {"type":"Literal","value":1,"raw":"1","start":1,"end":2},
+                            {"start":3,"end":3}
```

Comment:
The correct behavior is to not emit an element for void nodes. The AST should look like this:

````json
{
  "type": "Program",
  "start": 0,
  "end": 4,
  "body": [
    {
      "type": "ExpressionStatement",
      "start": 0,
      "end": 4,
      "expression": {
        "type": "ArrayExpression",
        "start": 0,
        "end": 4,
        "elements": [
          {
            "type": "Literal",
            "start": 1,
            "end": 2,
            "value": 1,
            "raw": "1"
          }
        ]
      }
    }
  ]
}
````

Please also "pretty-print" the JSON, f.e. using `python -mjson.tool`.